docs/proposals/override-values-inside-configurations-of-different-data-formats/README.md

```diff
@@ -0,0 +1,415 @@
++ Support native operations for data format that has its own important operations. XML with operations to modify its namespace, tag and attribute.
+### Non-Goals
++ Support every operation of different data formats.
++ Support all data formats.
+## Proposal
+### User Stories (Optional)
+#### Story 1
+As an administrator and developer, I want to update specific fields within complex data structures in resources to customize application behavior without replacing the entire configuration, ensuring that my changes are minimal and targeted.
+### Notes/Constraints/Caveats (Optional)
+### Risks and Mitigations
+## Design Details
+### API Change
+```go
+type Overriders struct {
+	...
+  // JSONPlaintextOverrider represents the rules dedicated to handling json object overrides
+	// +optional
+	JSONPlaintextOverrider []JSONPlaintextOverrider `json:"jsonPlaintextOverrider,omitempty"`
+	// YAMLPlaintextOverrider represents the rules dedicated to handling yaml object overrides
+	// +optional
+	YAMLPlaintextOverrider []YAMLPlaintextOverrider `json:"yamlPlaintextOverrider,omitempty"`
+}
+
+type JSONPlaintextOverrider struct {
+  // Path indicates the path of target field
+  Path string `json:"path"`
+  // JSONPatch represents json patch rules defined with plaintext overriders.
+  JSONPatch []JSONPatch `json:"jsonPatch"`
+  // MergeValue t represents the object value to be merged into the object.
+  MergeValue apiextensionsv1.JSON `json:"mergeValue"`
+  // MergeRawValue represents the raw, original format data (e.g., YAML, JSON) to be merged into the object.
+  MergeRawValue string `json:"mergeRawValue"`
+}
+
+type JSONPatch  struct {
+	// Path indicates the path of target field
+	Path string `json:"path"`
+	// From indicates the path of original field when operator is move and copy
+  From string `json:"from"`
+	// Operator indicates the operation on target field.
+	// Available operators are: add, replace and remove.
+	// +kubebuilder:validation:Enum=add;remove;replace;move;test;copy
+	Operator JSONPatchOperator `json:"operator"`
+	// Value to be applied to target field.
+	// Must be empty when operator is Remove.
+	// +optional
+	Value apiextensionsv1.JSON `json:"value,omitempty"`
+}
+
+// PatchOperator is the set of operators that can be used in an overrider.
+type JSONPatchOperator string
+
+// These are valid patch operators.
+const (
+	PatchOpAdd     JSONPatchOperator = "add"
+	PatchOpRemove  JSONPatchOperator = "remove"
+	PatchOpReplace  JSONPatchOperator = "replace"
+	PatchOpMove  JSONPatchOperator = "move"
+	PatchOpTest  JSONPatchOperator = "test"
+	PatchOpCopy  JSONPatchOperator = "copy"
+)
+
+type YAMLPlaintextOverrider struct {
+  // Path indicates the path of target field
+  Path string `json:"path"`
+  // JSONPatch represents json patch rules defined with plaintext overriders.
+  JSONPatch []JSONPatch `json:"jsonPatch"`
```

Comment:
Here, using `jsonPatch` doesn't seem very reasonable.

Reply:
Why? Hard to write the Patch in yaml?